examples/common/metrics_foo_library/CMakeLists.txt

```diff
@@ -1,6 +1,12 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+if(DEFINED OPENTELEMETRY_BUILD_DLL)
```

Comment:
This is not a problem of this PR, but I think we should use `if(OPENTELEMETRY_BUILD_DLL)` instead of `if(DEFINED OPENTELEMETRY_BUILD_DLL)`. And use `target_compile_definitions(opentelemetry_api INTERFACE OPENTELEMETRY_BUILD_IMPORT_DLL)` instead of `add_definitions(-DOPENTELEMETRY_BUILD_IMPORT_DLL)`, so we can exporting this definition to cmake package.